python3*Packages.dasbus tests failing

### Steps To Reproduce
Steps to reproduce the behavior:
1. build *python310Packages.dasbus* or package depending on it like nwg-panel

### Build log
```
======================================================================
FAIL: test_async_calls (tests.test_unix.DBusUnixExampleTestCase)
Test DBus async calls with fds.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/dasbus-1.7/tests/test_unix.py", line 420, in test_async_calls
    self._run_test()
  File "/build/dasbus-1.7/tests/lib_dbus.py", line 194, in _run_test
    self.assertEqual(client.exitcode, 0, msg)
AssertionError: 1 != 0 : _call_hello_async has finished with 1

======================================================================
FAIL: test_properties (tests.test_unix.DBusUnixExampleTestCase)
Test DBus properties with fds.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/dasbus-1.7/tests/test_unix.py", line 457, in test_properties
    self._run_test()
  File "/build/dasbus-1.7/tests/lib_dbus.py", line 194, in _run_test
    self.assertEqual(client.exitcode, 0, msg)
AssertionError: 1 != 0 : _set_pipes has finished with 1

======================================================================
FAIL: test_signals (tests.test_unix.DBusUnixExampleTestCase)
Test DBus signals with fds.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/dasbus-1.7/tests/test_unix.py", line 481, in test_signals
    self._run_test()
  File "/build/dasbus-1.7/tests/lib_dbus.py", line 194, in _run_test
    self.assertEqual(client.exitcode, 0, msg)
AssertionError: 1 != 0 : _trigger_signal has finished with 1

======================================================================
FAIL: test_sync_calls (tests.test_unix.DBusUnixExampleTestCase)
Test DBus sync calls with fds.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/dasbus-1.7/tests/test_unix.py", line 398, in test_sync_calls
    self._run_test()
  File "/build/dasbus-1.7/tests/lib_dbus.py", line 194, in _run_test
    self.assertEqual(client.exitcode, 0, msg)
AssertionError: 1 != 0 : _call_hello_sync has finished with 1

----------------------------------------------------------------------
Ran 232 tests in 679.212s

```
Full log: 

### Additional context
Wanted to ask for help here before asking upstream, seems to be specific as not reported or updated recently from what I've found.

Builds/seems to work fine with `doTest = false" but not the preferred way avoidable.

### Notify maintainers
@fortuneteller2k

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.19, NixOS, 23.05 (Stoat), 23.05.20230126.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/run/nixpkgs`
```

--- a/pkgs/development/python-modules/dasbus/default.nix
+++ b/pkgs/development/python-modules/dasbus/default.nix
@@ -1,16 +1,20 @@
-{ lib, buildPythonPackage, fetchPypi, pygobject3, dbus }:
+{ lib, buildPythonPackage, fetchFromGitHub, pygobject3, dbus, hatchling, pytestCheckHook }:
 
 buildPythonPackage rec {
   pname = "dasbus";
-  version = "1.7";
+  version = "unstable-11-10-2022";
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-qIUNhBrf6O5fe7n4LPRJq5tJUNwGM4lwcXGODQA2tvY=";
+  src = fetchFromGitHub {
+    owner = "rhinstaller";
+    repo = pname;
+    rev = "64b6b4d9e37cd7e0cbf4a7bf75faa7cdbd01086d";
+    hash = "sha256-TmhhDrfpP+nUErAd7dUb+RtGBRtWwn3bYOoIqa0VRoc=";
   };
 
+  nativeBuildInputs = [ hatchling ];
   propagatedBuildInputs = [ pygobject3 ];
-  nativeCheckInputs = [ dbus ];
+  nativeCheckInputs = [ dbus pytestCheckHook ];
 
   meta = with lib; {
     homepage = "https://github.com/rhinstaller/dasbus";

